value.go

```diff
@@ -405,7 +405,11 @@ type Decimal int64
 // DecimalPrecision is the precision of a Decimal.
 const DecimalPrecision = 10000
 
-func newDecimalValue(s string) (Decimal, error) {
+// ParseDecimal takes a string representation of a decimal number and converts it into a Decimal type.
```

Comment:
I actually think just the first line is the only one that is necessary.  (Thanks for all the detail though, but it seems a bit much to me!)